### PR TITLE
docs: Added command to build and deploy in olm enabled kubernetes

### PR DESCRIPTION
**Pull request by rishabh625**
Signed-off-by: rishabh625 <rishabhmishra625@gmail.com>

**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

> /kind bug
> /kind chore
> /kind cleanup
> /kind failing-test
> /kind enhancement
/kind documentation
> /kind code-refactoring

**What does this PR do / why we need it**:

**Have you updated the necessary documentation?**

* [x] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -238,3 +238,15 @@ catalog-build: opm ## Build a catalog image.
...
+KUBECTL = $(shell which kubectl)
+build-deploy : docker-build docker-push clear-bundle bundle bundle-build bundle-push catalog-build catalog-push
+	$(shell cat deploy/catalog_source.yaml| sed 's,image: .*,image: $(CATALOG_IMG),' > deploy/catalog_source_temp.yaml)
+	$(KUBECTL) create -n olm -f deploy/catalog_source_temp.yaml
+	rm -rf deploy/catalog_source_temp.yaml
+	$(KUBECTL) create ns argocd
+	$(KUBECTL) create -n argocd -f deploy/operator_group.yaml
+	$(KUBECTL) create -n argocd -f deploy/subscription.yaml
+clear-bundle :
+	rm -fr bundle/
```
Comment (wtam2018) [inline Makefile:252]: I am not sure we need to add a new target to combine build and deploy.  Please do `make help` and see if there are existing targets that suit your need. You will notice that targets are organized by `Dev`, `Build`, and `Deploy`. We are trying to keep the Makefile clean.

Comment (rishabh625) [reply Makefile:252]: It think it would be helpful for testing operator locally , by building and deploying on local cluster.

Comment (wtam2018) [reply Makefile:252]: Doesn't `make bundle` generates `deploy/catalog_source.yaml` with image = `$CATALOG_IMG` already?

Comment (rishabh625) [reply Makefile:252]: It creates bundle and and CSV , whereas this is convenient command to invoke all commands required to build and test locally, but coming to your previous comment this doesn't keep makefile cleaner , where each command is concerned with one thing build or deploy , we can close this.

**Pull request closed without merging**